torchio/transforms/augmentation/intensity/random_simulation.py

```diff
@@ -0,0 +1,172 @@
+
+from typing import Union, Tuple, Optional, Dict, List
+import torch
+from ....torchio import DATA, TypeData, TypeRangeFloat, TypeNumber, AFFINE, INTENSITY, TYPE, LABEL
+from ....data.subject import Subject
+from ....data.image import Image
+from .. import RandomTransform
+
+MEAN_RANGE = (0.1, 0.9)
+STD_RANGE = (0.01, 0.1)
+
+
+class RandomSimulation(RandomTransform):
+    r"""Generate a random image from label maps.
+
+    Args:
+        label_keys: Keys designating the label images in the sample that
+            will be used to generate the new image.
+            If a single string is given, the corresponding label map is expected
+            to be a label encoding map.
+            Default will use all images labeled as label in the sample.
+        image_key: Key to which the new volume will be saved.
```

Comment:
we would like to have the possibility to make a mixture, of pure synthetic label image and original volume for inpainting. (In our use case we have precise GM WM and CSF, that we would like to add on realistic Head and background

But It should be a more explicit name like inpainting_image_key. I do not understand the save use (if this is the name to save the result, it should be an independant parameter ... non?

Reply:
"Stored" will probably be more accurate than "saved", as the transform creates a new image, it must store it in the sample after its creation. This parameter corresponds to the key used to store the new image.

However, and indeed I think it was a bad idea, this key was also used to mix the original volume and synthetic one if this key corresponded to an already existing volume in the sample.

Reply:
What do you mean by inpainting? For me, it's like feeling holes with the background texture: http://math.univ-lyon1.fr/homes-www/masnou/fichiers/publications/survey.pdf

So you have multiple fractional label maps e.g. from different tissues stored in different images in the subject and you want to create the image from that, I guess. You don't want to create a label map with them using e.g. `argmax`? So this is actually more generic and complex than lab2im...

Reply:
> What do you mean by inpainting? For me, it's like feeling holes with the background texture: http://math.univ-lyon1.fr/homes-www/masnou/fichiers/publications/survey.pdf

Yes you're right, inpainting is not the right concept, filling holes should probably be the right one.

> So you have multiple fractional label maps e.g. from different tissues stored in different images in the subject and you want to create the image from that, I guess. You don't want to create a label map with them using e.g. argmax? So this is actually more generic and complex than lab2im...

Taking the `argmax` would be the binary way of doing. With PVs you want to take the weighted sums over your labels.

Reply:
> ? So this is actually more generic and complex than lab2im..

yes, we want to follow the lab2im work, but we also add an important improvement : correctly handeling of the Partial Volume (PV) effect 
They main difference is there, instead of dealing with binary lablel we handel PV labels. It is more close to what MRI acquisition is doing 
(in their work the use bluring to simulate the PV, althought here we use the exact generic model : a weighted sum .